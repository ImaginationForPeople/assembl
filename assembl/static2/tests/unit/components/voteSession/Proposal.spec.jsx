import React from 'react';
import renderer from 'react-test-renderer';

import Proposal from '../../../../js/app/components/voteSession/proposal';
import * as fakeData from './fakeData';
import '../../../helpers/setupTranslations';

describe('Proposal component', () => {
  const voteForProposalSpy = jest.fn();
  const voteForProposalGaugeSpy = jest.fn();
  it('should match Proposal snapshot', () => {
    const { remainingTokensByCategory, userGaugeVotes, userTokenVotes, tokenVoteModule, textGaugeModule } = fakeData;
    const props = {
      description: 'Try to override the CSS transmitter, maybe it will program the solid state program!',
      id: 'my-proposal',
      modules: [tokenVoteModule, textGaugeModule],
      remainingTokensByCategory: remainingTokensByCategory,
      seeCurrentVotes: true,
      title: 'Use the online USB transmitter, then you can reboot the cross-platform application!',
      userGaugeVotes: userGaugeVotes,
      userTokenVotes: userTokenVotes,
      voteForProposal: voteForProposalSpy,
      voteForProposalGauge: voteForProposalGaugeSpy
    };
    const rendered = renderer.create(<Proposal {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});