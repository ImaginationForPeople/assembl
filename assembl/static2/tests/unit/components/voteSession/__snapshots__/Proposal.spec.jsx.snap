// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proposal component should match Proposal snapshot 1`] = `
<div
  className="theme-box"
>
  <div
    className="proposal row"
  >
    <div
      className="margin-s col-md-5 col-xs-12"
    >
      <h3
        className="proposal-title dark-title-3"
      >
        Use the online USB transmitter, then you can reboot the cross-platform application!
      </h3>
      <div
        className="text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Try to override the CSS transmitter, maybe it will program the solid state program!",
          }
        }
      />
    </div>
    <div
      className="proposal-vote-modules col-md-6 col-md-offset-1 col-xs-12"
    >
      <div
        className="margin-m"
      >
        <p>
          If we generate the feed, we can get to the AGP driver through the primary ADP interface!
        </p>
        <div
          className="tokens-line"
        >
          <p>
            Positive
          </p>
          <div
            className="tokens"
          >
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#226622"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <span
                className="assembl-icon-delete grey"
              />
            </button>
          </div>
        </div>
        <div
          className="tokens-line"
        >
          <p>
            Negative
          </p>
          <div
            className="tokens"
          >
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#771122"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#771122"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#771122"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#771122"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#771122"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#771122"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#771122"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#771122"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#771122"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="admin-icons btn btn-default"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                height={28}
                role="img"
                version="1.1"
                viewBox="0 0 28 28"
                width={28}
                x="0"
                xmlSpace="preserve"
                y="0"
              >
                <circle
                  cx={14}
                  cy={14}
                  fill="none"
                  r={13}
                  stroke="#771122"
                  strokeWidth={2}
                />
              </svg>
            </button>
            <button
              className="btn btn-default"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <span
                className="assembl-icon-delete grey"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="gauge-vote-for-proposal"
      >
        <p>
          I'll generate the mobile SMS bus, that should microchip the TCP matrix!
        </p>
        <div
          className="rc-slider"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <div
            className="rc-slider-rail"
            style={
              Object {
                "backgroundColor": "#E6E5F4",
                "height": "12px",
              }
            }
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "height": "12px",
                "left": "0%",
                "visibility": "visible",
                "width": "NaN%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
            role="slider"
            style={
              Object {
                "border": "none",
                "boxShadow": "none",
                "height": "0",
                "left": "NaN%",
                "marginTop": "-20px",
                "width": "0",
              }
            }
            tabIndex={0}
          >
            <svg
              className="pointer"
              height="46px"
              style={
                Object {
                  "cursor": "-webkit-grab",
                  "marginLeft": "initial",
                  "marginTop": "-15px",
                }
              }
              version="1.1"
              viewBox="0 0 31 46"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                transform="translate(-772.000000, -4682.000000)"
              >
                <g
                  transform="translate(-2.000000, 689.000000)"
                >
                  <g
                    transform="translate(2.000000, 2971.000000)"
                  >
                    <g
                      transform="translate(101.000000, 1022.000000)"
                    >
                      <g
                        transform="translate(671.000000, 0.000000)"
                      >
                        <path
                          d="M15.2198519, 0 C6.81196296, 0 0,6.81196296 0, 15.2198519 C0, 21.7436296 4.12185185, 27.2593333 9.88383333, 29.4293333 L11.4148889, 34.0787593 L15.2192778, 45.6601296 L19.0248148, 34.0787593 L20.5552963, 29.4293333 C26.3195741, 27.2593333 30.4397037, 21.7436296 30.4397037, 15.2198519 C30.4397037, 6.81196296 23.6202778, 0 15.2198519, 0 Z M14.9672593, 19.786037 C12.4459259, 19.786037 10.4010741, 17.7411852 10.4010741, 15.2198519 C10.4010741, 12.6990926 12.4459259, 10.6542407 14.9672593, 10.6542407 C17.4880185, 10.6542407 19.5328704, 12.6990926 19.5328704, 15.2198519 C19.5328704, 17.7406111 17.4880185, 19.786037 14.9672593, 19.786037 Z"
                          id="Shape"
                        />
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </div>
          <div
            className="rc-slider-mark"
          />
        </div>
        <input
          name="vote-for-proposal-my-proposal-vote-specification-text-gauge-module"
          type="hidden"
          value={0}
        />
      </div>
    </div>
  </div>
  <div
    className="expand-result"
  >
    <div
      className="expand-row background-grey row"
    >
      <div
        className="center col-md-12 col-xs-12"
      >
        <button
          className="link-button btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className={undefined}
            style={undefined}
          >
            Show votes of the community
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
